relays/client-substrate/src/transaction_tracker.rs

```diff
@@ -213,150 +213,102 @@ async fn watch_transaction_status<C: Chain, S: Stream<Item = TransactionStatusOf
 mod tests {
 	use super::*;
 	use crate::test_chain::TestChain;
-	use futures::FutureExt;
+	use futures::{FutureExt, SinkExt};
+	use relay_utils::TransactionTracker as _;
 	use sc_transaction_pool_api::TransactionStatus;
 
-	fn tx_hash() -> HashOf<TestChain> {
-		Default::default()
-	}
+	async fn on_transaction_status(
+		status: TransactionStatus<HashOf<TestChain>, HashOf<TestChain>>,
+	) -> Option<TrackedTransactionStatus> {
+		let (mut sender, receiver) = futures::channel::mpsc::channel(1);
+		let tx_tracker = TransactionTracker::<TestChain>::new(
+			Duration::from_secs(0),
```

Comment:
Nit: Sorry, just one more thing. We could also test that it waits on certain `TransactionStatus` values. I think we could use `Duration::from_secs(1),` and time the function execution and check that it takes longer than 1 sec. However it would add an overhead to the time it takes to run the unit tests. Not sure how problematic this would be.

Reply:
I've made this `do_wait` method, which imo is a good compromise between black box testing and testing some internals (`InvalidationStatus`)